### PR TITLE
ModulesGarden

### PR DESCRIPTION
**Pull request by max-mi**
Empty SSO Token Validate
Try to create log file if it does not exist

### CODE REVIEW

File: src/js/zendesk/validation.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Validation.addAllThese([
+    ['validate-zendesk-sso-token', 'Token cannot be empty', function (v) {
+            if ($('zendesk_sso_enabled').getValue() === '1') {
```
Comment (jwswj) [inline src/js/zendesk/validation.js]: excessive indentation here should be:

``` js
Validation.addAllThese([
    ['validate-zendesk-sso-token', 'Token cannot be empty', function(v) {
        if ($('zendesk_sso_enabled').getValue() === '1') {
            return !Validation.get('IsEmpty').test(v);
        } else {
            return true;
        }

    }],
    ['validate-zendesk-sso-frontend-token', 'Token cannot be empty', function(v) {
        if ($('zendesk_sso_frontend_enabled').getValue() === '1') {
            return !Validation.get('IsEmpty').test(v);
        } else {
            return true;
        }

    }]
]);
```

File: src/js/zendesk/validation.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+                return !Validation.get('IsEmpty').test(v);
+            } else {
+                return true;
+            }
+
+        }],
+    ['validate-zendesk-sso-frontend-token', 'Token cannot be empty', function (v) {
```
Comment (jwswj) [inline src/js/zendesk/validation.js:9]: How does this work for non english versions of the integration?

File: src/app/code/community/Zendesk/Zendesk/controllers/Adminhtml/ZendeskController.php [hunk 3]
```diff
@@ -364,9 +364,10 @@ public function autocompleteAction()
     public function logAction()
     {
         $path = Mage::helper('zendesk/log')->getLogPath();
-        
+
         if(!file_exists($path)) {
             Mage::getSingleton('adminhtml/session')->addError(Mage::helper('zendesk')->__('The Zendesk log file has not been created. Check to see if logging has been enabled.'));
+            file_put_contents(" ",$path);
```
Comment (jwswj) [inline src/app/code/community/Zendesk/Zendesk/controllers/Adminhtml/ZendeskController.php]: What was creating the log in the past?
Shouldn't the log message above be changed to "Creating a Zendesk log file"?

Comment (max-mi) [inline src/app/code/community/Zendesk/Zendesk/controllers/Adminhtml/ZendeskController.php]: Fixed, added additional condition.

File: src/js/zendesk/validation.js (hunk 2 above)
Comment (max-mi) [inline src/js/zendesk/validation.js:9]: Added JS Localisation System

File: src/app/design/adminhtml/default/default/layout/zendesk.xml [hunk 4]
```diff
@@ -22,6 +22,12 @@
             <action method="addCss">
                 <stylesheet>zendesk/zendesk.css</stylesheet>
             </action>
+            <action method="addJs">
+                <script>zendesk/validation.js</script>
+            </action>
+        </reference>
+        <reference name="content">
+            <block type="core/template" name="test" template="zendesk/translations.phtml"></block>
```
Comment (jwswj) [inline src/app/design/adminhtml/default/default/layout/zendesk.xml]: why is the `name="test"`? is a name required? shouldn't it be translation?

File: src/app/design/adminhtml/default/default/template/zendesk/translations.phtml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+?>
+<script>
+Translator.add('Token cannot be empty', '<?php echo $this->__('Token cannot be empty'); ?>');
```
Comment (jwswj) [inline src/app/design/adminhtml/default/default/template/zendesk/translations.phtml]: This should be indented.

File: src/app/design/adminhtml/default/default/layout/zendesk.xml (hunk 4 above)
Comment (max-mi) [inline src/app/design/adminhtml/default/default/layout/zendesk.xml]: Name isn't required here.

**Comment by jwswj** [comment]
:+1:

**Pull request merged**